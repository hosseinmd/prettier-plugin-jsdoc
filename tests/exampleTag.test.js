const prettier = require("prettier");

function subject(code, options = {}) {
  return prettier.format(code, {
    parser: "babel",
    plugins: ["."],
    jsdocSpaces: 1,
    ...options,
  });
}

test("Example javascript code", () => {
  const result = subject(`
/**
* @examples
*   var one = 5
*   var two = 10

     const resolveDescription = formatDescription(tag, description, tagString, a);
*
*   if(one > 2) { two += one

 }

* @undefiendTag
* @undefiendTag {number} name des
*/
const testFunction = (text, defaultValue, optionalNumber) => true
`);

  expect(result).toMatchSnapshot();
  expect(subject(result)).toMatchSnapshot();
});

test("empty example", () => {
  const Result2 = subject(`/**
 * single line description
 * @example
 */`);

  const Result3 = subject(`/**
 * single line description
 * @return {Boolean} Always true
 * @example
 */`);

  expect(Result2).toMatchSnapshot();
  expect(Result3).toMatchSnapshot();
});

test("examples Json", () => {
  const options = {
    jsdocKeepUnParseAbleExampleIndent: true,
  };
  const Result1 = subject(
    `/**
 * @example
 * {testArr: [
 *     1,
 *     2,
 *   ]
 *  }
 */`,
    options,
  );

  const Result2 = subject(
    `/**
 * @example
 *
 * [{
 *   foo: 1,
 *   foo: 2,
 *   foo: 9,
 * }, {
 *   bar: 1,
 *   bar: 5
 * }]
 */`,
    options,
  );
  expect(Result1).toMatchSnapshot();
  expect(Result2).toMatchSnapshot();
});

test("Example start by xml tag", () => {
  const result = subject(`
  /**
   * @example <caption>TradingViewChart</caption>;
   *
   * export default Something
   */
`);

  expect(result).toMatchSnapshot();

  const result1 = subject(`
  /**
   * @example <caption>TradingViewChart</caption>
   *
   * function Something(){
   *   return <caption>TradingViewChart</caption>
   * }
   * export default Something
   */
`);

  expect(result1).toMatchSnapshot();
});

test("example json ", () => {
  const result = subject(`
  /**
   * @example {
   *   '0%': '#afc163',
   *   '25%': '#66FF00',
   *   '50%': '#00CC00',     // ====>  linear-gradient(to right, #afc163 0%, #66FF00 25%,
   *   '75%': '#009900',     //         #00CC00 50%, #009900 75%, #ffffff 100%)
   *   '100%': '#ffffff'
   * }
   * @description
   * Then this man came to realize the truth:
   * Besides six pence, there is the moon.
   * Besides bread and butter, there is the bug.
   * And...
   * Besides women, there is the code.
   */
`);

  expect(result).toMatchSnapshot();
});

test("example should be same after few time format ", () => {
  const result = subject(`
  /**
   * @example <caption>with selector</caption>
   *   const $ = ccashio.test(\`
   *     <div id=test>
   *       <p>Hello</p>
   *       <b><p>World</p></b>
   *     </div>
   *   \`);
  */
`);

  const result2 = subject(result);
  const result3 = subject(result2);

  expect(result).toEqual(result2);
  expect(result).toEqual(result3);
});

test("example with unknown language should be same after few time format ", () => {
  const options = { jsdocKeepUnParseAbleExampleIndent: true };

  // the `'comment': { ... }` line makes it invalid JS
  const result = subject(
    `
  /**
   * @example
   *   Prism.languages['css-with-colors'] = Prism.languages.extend('css', {
   *       // Prism.languages.css already has a 'comment' token, so this token will overwrite CSS' 'comment' token
   *       // at its original position
   *       'comment': { ... },
   *       // CSS doesn't have a 'color' token, so this token will be appended
   *       'color': /\\b(?:red|green|blue)\\b/
   *   });
   *
  */
`,
    options,
  );

  expect(result).toMatchSnapshot();

  const result2 = subject(result, options);
  const result3 = subject(result2, options);

  expect(result).toEqual(result2);
  expect(result).toEqual(result3);
});
