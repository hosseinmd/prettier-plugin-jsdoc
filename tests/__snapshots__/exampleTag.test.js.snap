// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example javascript code 1`] = `
"/**
 * @example
 *   var one = 5;
 *   var two = 10;
 *
 *   const resolveDescription = formatDescription(
 *     tag,
 *     description,
 *     tagString,
 *     a
 *   );
 *
 *   if (one > 2) {
 *     two += one;
 *   }
 *
 * @undefiendTag
 * @undefiendTag {number} name des
 */
const testFunction = (text, defaultValue, optionalNumber) => true;
"
`;

exports[`Example javascript code 2`] = `
"/**
 * @example
 *   var one = 5;
 *   var two = 10;
 *
 *   const resolveDescription = formatDescription(
 *     tag,
 *     description,
 *     tagString,
 *     a
 *   );
 *
 *   if (one > 2) {
 *     two += one;
 *   }
 *
 * @undefiendTag
 * @undefiendTag {number} name des
 */
const testFunction = (text, defaultValue, optionalNumber) => true;
"
`;

exports[`Example start by xml tag 1`] = `
"/**
 * @example <caption>TradingViewChart</caption>
 *   export default Something;
 */
"
`;

exports[`Example start by xml tag 2`] = `
"/**
 * @example <caption>TradingViewChart</caption>
 *   function Something() {
 *     return <caption>TradingViewChart</caption>;
 *   }
 *   export default Something;
 */
"
`;

exports[`empty example 1`] = `
"/** Single line description */
"
`;

exports[`empty example 2`] = `
"/**
 * Single line description
 *
 * @returns {Boolean} Always true
 */
"
`;

exports[`example json  1`] = `
"/**
 * Then this man came to realize the truth: Besides six pence, there is the
 * moon. Besides bread and butter, there is the bug. And... Besides women,
 * there is the code.
 *
 * @example
 *   {
 *     \\"0%\\": \\"#afc163\\",
 *     \\"25%\\": \\"#66FF00\\",
 *     \\"50%\\": \\"#00CC00\\", // ====>  linear-gradient(to right, #afc163 0%, #66FF00 25%,
 *     \\"75%\\": \\"#009900\\", //         #00CC00 50%, #009900 75%, #ffffff 100%)
 *     \\"100%\\": \\"#ffffff\\"
 *   }
 */
"
`;

exports[`examples Json 1`] = `
"/**
 * @example
 *   { \\"testArr\\": [1, 2] }
 */
"
`;

exports[`examples Json 2`] = `
"/**
 * @example
 *   [
 *     {
 *       foo: 1,
 *       foo: 2,
 *       foo: 9,
 *     },
 *     {
 *       bar: 1,
 *       bar: 5,
 *     },
 *   ];
 */
"
`;
